### PR TITLE
Allow default improver to improve without affected packages

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

File: vulnerabilities/tests/test_default_improver.py [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+        summary="Test summary",
+        references=[
+            Reference(
+                url="https://www.example.com/",
+            )
+        ],
+        affected_packages=[],
+    )
+    default_improver = DefaultImprover()
+    assert [
```
Comment (pombredanne) [inline vulnerabilities/tests/test_default_improver.py:42]: create an expected var. and use the order results == expected.

**Review by pombredanne — CHANGES_REQUESTED** [review]
LGTM... but consider using file-driven tests for easier maintenance and updates. You my the nginx test branch merged first.

File: vulnerabilities/tests/test_default_improver.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    default_improver = DefaultImprover()
+    response_file = os.path.join(TEST_DATA, f"alpine-input.json")
+    with open(response_file, "r") as f:
+        advisory_data = json.load(f)
+    expected_file = os.path.join(TEST_DATA, f"alpine-expected.json")
+    result = [
+        data.to_dict()
+        for data in list(default_improver.get_inferences(AdvisoryData.from_dict(advisory_data)))
+    ]
+    if regen:
```
Comment (pombredanne) [inline vulnerabilities/tests/test_default_improver.py:119]: We will need to switch to this kind of approach IMHO, e.g. using commoncode plus this test helper: `util_tests.check_results_against_json()` here and below.
https://github.com/nexB/vulnerablecode/blob/d1f21983d39592b7470a1db038d66c0031053b30/vulnerabilities/tests/test_nginx.py#L232

**Review by pombredanne — APPROVED** [review]
Thanks!

File: vulnerabilities/tests/test_default_improver.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    assert result == expected
+
+
+def test_default_improver_with_alpine(regen=VULNERABLECODE_REGEN_TEST_FIXTURES):
+    """
+    Test that DefaultImprover.get_inferences() returns an empty list when given an advisory with alpine advisories
+    """
+    default_improver = DefaultImprover()
+    response_file = os.path.join(TEST_DATA, f"alpine-input.json")
+    with open(response_file, "r") as f:
```
Comment (pombredanne) [inline vulnerabilities/tests/test_default_improver.py:109]: "r" is always the default FWIW, so not needed in general

File: vulnerabilities/improvers/default.py [hunk 4]
```diff
@@ -50,15 +50,25 @@ def interesting_advisories(self) -> QuerySet:
...
-            yield Inference(
-                aliases=advisory_data.aliases,
-                confidence=MAX_CONFIDENCE,
-                summary=advisory_data.summary,
-                affected_purls=affected_purls,
-                fixed_purl=fixed_purl,
-                references=advisory_data.references,
+
+        if not advisory_data:
+            return None
```
Comment (Hritik14) [inline vulnerabilities/improvers/default.py:55]: Return type is an iterable

**Pull request merged**